### PR TITLE
[Spec] Remove dead `padded_static_len` and stale `SGLANG_ENABLE_SPEC_V2` references

### PR DESCRIPTION
**Pull request by hnyls2002**
Stacks on #30977. Two dead-residue removals from the retired spec V1 / SpecV2-flag era:

- **Remove dead `padded_static_len`**: no production writer other than the constant `-1` since the spec V1 workers were removed (#25464). The padded-indexing branch in `LogitsProcessor` is unreachable, and the `ForwardBatch` / `LogitsMetadata` fields plus the graph-runner pass-throughs are dead plumbing. The only non-`-1` writer was an attention-unittest kit whose batches never reach the logits processor.
- **Remove stale `SGLANG_ENABLE_SPEC_V2` references**: the env var was removed (spec always runs the V2 worker; the launch-time removal notice stays), but ~29 tests still set it as a no-op and `docs_new` deployment snippets still taught users to export it.

---
### CI States

Latest PR Test (Base): :no_entry_sign: [Run #29281901175](https://github.com/sgl-project/sglang/actions/runs/29281901175)
Latest PR Test (Extra): :x: [Run #29281900900](https://github.com/sgl-project/sglang/actions/runs/29281900900)

### CODE REVIEW

**Comment by hnyls2002** [comment]
/rerun-test test_spec_eagle.py test_eagle_draft_cuda_graph_runner.py test_tbo.py
